src/CompositeRheologies.jl

```diff
@@ -794,39 +1005,235 @@ end
+    
+    x    = @MVector zeros(_T, n)
+    x[1] = τ_initial
+
+    j = 1;
+    for i=1:N
+
+        if is_plastic[i] && is_par[i]
+            # parallel plastic element
+            j=j+2
+            x[j] = τ_initial    # τ_plastic initial guess     
+            j += 1
+            x[j] = p_initial    # initial guess for pressure
+        end
+        if is_plastic[i] && !is_par[i]
+            # parallel plastic element
+            j += 1
+            x[j] = 0    # λ̇  
+            j += 1
+            x[j] = p_initial    # initial guess for pressure
+        end
+            
+    end
+#=    
+        if is_plastic[i] 
+           # plastic element 
+           j += 2
+           x[j] = 0    # λ̇  
+
+            j += 1
+            x[j] = p_initial    # initial guess for pressure
+        end
+        if is_par[i] 
+            j += 1
+            x[j] = τ_initial    # τ_plastic initial guess  
+        end
+       # if is_vol[i]
+       #     # dilatant element
+       #     j += 1
+       #     x[j] = p_initial    # initial guess for pressure
+       # end
+=#
+#end
+
+    r = @MVector zeros(_T,n);
+    J = @MMatrix zeros(_T, n,n)   # size depends on # of plastic elements
+    
+    # Local Iterations
+    iter = 0
+    ϵ = 2 * tol
+    while (ϵ > tol) && (iter < max_iter)
+        iter += 1
+
+        τ   = x[1]
+        λ   = x[2]
+        P   = x[n]
+        
+        args = merge(args, (τII=τ, P=P, λ=λ))    # update (to compute yield function)
+
+        # Update part of jacobian related to serial, non-plastic, elements
+        r[1]   = εII_total - compute_εII_elements(c,τ,args)     
+        J[1,1] = dεII_dτII_elements(c,τ,args);               
+        
+        r[n]   = εvol_total - compute_εvol_elements(c,P,args)     
+        J[n,n] = dεvol_dp(c,P,args);               
+        
+        # Add contributions from plastic elements
+        fill_J_plastic!(J, r, x, c, args)
+
+        #@show x r J
+        
+        # update solution
+        dx  = J\r 
+        x .+= dx   
+        
+        ϵ    = sum(abs.(dx)./(abs.(x .+ 1e-9)))
+        verbose && println(" iter $(iter) $ϵ F=$(r[2]) τ=$(x[1]) λ=$(x[2]) P=$(x[3])")
+    end
+    verbose && println("---")
+    if (iter == max_iter)
+        error("iterations did not converge")
+    end
+
+    return (x...,)
+end
+
 @generated function ∂Q∂τII(
-    v::Parallel{T, N,  Nplast, is_plastic}, τ::_T
+    v::Parallel{T, N,  Nplast, is_plastic}, τ::_T, args
 ) where {_T,T, N,  Nplast, is_plastic}
     quote
         Base.@_inline_meta
-        Base.Cartesian.@nexprs $N i -> is_plastic[i] == true && return ∂Q∂τII(v[i],τ)
+        Base.Cartesian.@nexprs $N i -> is_plastic[i] == true && return ∂Q∂τII(v[i],τ, args)
+    end
+end
+
+@generated function ∂Q∂P(
+    v::Parallel{T, N,  Nplast, is_plastic}, P::_T, args
+) where {_T,T, N,  Nplast, is_plastic}
+    quote
+        Base.@_inline_meta
+        Base.Cartesian.@nexprs $N i -> is_plastic[i] == true && return ∂Q∂P(v[i],P, args)
```

Comment:
Add `return zero(_T)` at the end

Reply:
hmm, this doesn't actually seem to work (tests fail if I add this)